boards/imix/src/main.rs

```diff
@@ -347,16 +352,28 @@ pub unsafe fn main() {
         },
     );
 
-    let board_kernel = static_init!(kernel::Kernel, kernel::Kernel::new(&PROCESSES));
-
     let dynamic_deferred_call_clients =
-        static_init!([DynamicDeferredCallClientState; 5], Default::default());
+        static_init!([DynamicDeferredCallClientState; 6], Default::default());
     let dynamic_deferred_caller = static_init!(
         DynamicDeferredCall,
         DynamicDeferredCall::new(dynamic_deferred_call_clients)
     );
     DynamicDeferredCall::set_global_instance(dynamic_deferred_caller);
 
+    let sha = static_init!(
+        Sha256Software<'static>,
+        Sha256Software::new(dynamic_deferred_caller)
+    );
+    sha.initialize_callback_handle(dynamic_deferred_caller.register(sha).unwrap());
+
+    let checker = static_init!(AppCheckerSha256, AppCheckerSha256::new(sha));
```

Comment:
Why not pass the buffer in here and avoid the global mut in the kernel?

Reply:
The SHA256 checker is intended to be an example for testing rather than general use, so I cut this corner when testing and didn't fix it. I agree the buffer should be passed. Fixed.